src/pybind/mgr/dashboard/frontend/src/app/shared/directives/auth-storage.directive.spec.ts

```diff
@@ -0,0 +1,87 @@
+import { Component } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { Permissions } from '~/app/shared/models/permissions';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+import { configureTestBed } from '~/testing/unit-test-helper';
+import { AuthStorageDirective } from './auth-storage.directive';
+@Component({
+  template: `<div *cdAuthStorage="condition"></div>`
+})
+export class AuthStorageDirectiveTestComponent {
+  condition: string | string[] | object;
+}
+
+describe('AuthStorageDirective', () => {
+  let fixture: ComponentFixture<AuthStorageDirectiveTestComponent>;
+  let component: AuthStorageDirectiveTestComponent;
+  let getPermissionsSpy: jasmine.Spy;
+  const getDiv = () => fixture.debugElement.nativeElement.querySelector('div');
```

Comment:
This is a minor thing, but from a reviewers perspective (and plz keep in mind that code is written once but read many times), what I see is a test that looks for a div inside the component, but that's not very meaningful.

What about using an Angular template reference variable with a more meaningful name (`#permitted` or id=permitted or a class=permitted)?
```html
<div #permitted ...
```

And then:
```ts
el = fixture.debugElement.nativeElement;
...
expect(el.querySelector('#permitted')).toBeTruthy();
```